src/status_im/ui2/screens/chat/group_details/view.cljs

```diff
@@ -0,0 +1,106 @@
+(ns status-im.ui2.screens.chat.group-details.view
+  (:require [react-native.core :as rn]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.buttons.button :as quo2.button]
+            [status-im.ui2.screens.chat.group-details.style :as style]
+            [quo2.core :as quo2]
+            [quo2.components.icon :as icons]
+            [utils.re-frame :as rf]
+            [status-im.i18n.i18n :as i18n]
+            [status-im.ui.components.list.views :as list]
+            [status-im.ui2.screens.chat.components.contact-item.view :as contact-item]))
+
+(defn back-button []
+  [quo2.button/button {:type                :grey
+                       :size                32
+                       :width               32
+                       :style               {:margin-left 20}
+                       :accessibility-label "back-button"
+                       :on-press            #(do
+                                               (rf/dispatch [:navigate-back]))}
+   [icons/icon :i/arrow-left {:color (colors/theme-colors colors/neutral-100 colors/white)}]])
+
+(defn options-button []
+  [quo2.button/button {:type                :grey
+                       :size                32
+                       :width               32
+                       :style               {:margin-right 20}
+                       :accessibility-label "options-button"}
+   [icons/icon :i/options {:color (colors/theme-colors colors/neutral-100 colors/white)}]])
+
+(defn count-container [count]
+  [rn/view {:style (style/count-container)}
+   [quo2/text {:size   :label
+               :weight :medium
+               :style  {:text-align :center}} count]])
+
+(defn prepare-members [contacts admins]
+  (let [data (atom {:owner   {:title :Owner
```

Comment:
I think we use reagent/atom

Reply:
Good catch @ibrkhalil. From the implementation of `prepare-members`, it looks like it's using an atom in order to process data in an imperative loop using `doseq`, thus less functional. There's definitely a simpler way to do this in Clojure.

Reply:
@OmarBasem, @ibrkhalil, please, don't take this snippet too seriously, but as far as I've tested in a REPL, it behaves just as the original implementation. I had some fun with it ;) Omar, feel free to grab or adapt this implementation. While writing it, I felt the need for a unit test too.

I don't know if this is new info to any of you, but you guys might like the reading about https://clojure.org/reference/transients. For high performance code, it has far less overhead than creating temporary atoms for data processing, but by no means I would use them in this `prepare-members` function!!!

```clojure
(defn prepare-members
  [contact-addresses admins]
  (let [contacts (map (fn [address]
                        (assoc (rf/sub [:contacts/contact-by-address address])
                               :admin? (get admins address)))
                      contact-addresses)]
    (as-> {} $
      (when-let [data (filter :admin? contacts)]
        (assoc $ :owner {:title :Owner :data admins}))
      (when-let [data (filter #(and (not (:admin? %)) (:online? %)) contacts)]
        (assoc $ :online {:title :Online :data data}))
      (when-let [data (filter #(and (not (:admin? %)) (not (:online? %))) contacts)]
        (assoc $ :offline {:title :offline :data data}))
      (vals $))))
```

Reply:
Thanks for the info @ilmotta. I will take a look into it. 
And thanks for the snippet. It works correctly, and looks much better :)